Fixes for C12 MOM6

mathomp4 opened the pull request:
This PR updates the setup scripts for the C12 MOM6 setup that @sanAkel and @yvikhlya have developed. It includes:

- Fixes some csh logic for making the tile names
- Default C12 MOM6 as 1x6 atmos/3x2 ocean. Note that C12 data ocean will still be 2x12 by default

sanAkel commented:
>NOTE2: At this point I think you can only run C12 MOM6 with my TinyBCs. I believe @yvikhlya needs to copy some files to a new /discover/nobackup/yvikhlia/coupled/Forcings/MOM6/CF0012x6C_TM0072xTM0036 directory that I can provide.

I think this has happened. 
@mathomp4 would you please check if ALL still works at your end?

mathomp4 commented:
@sanAkel I still do not see a `CF0012x6C_TM0072xTM0036` directory in:
```
/discover/nobackup/yvikhlia/coupled/Forcings/MOM6/
```

I think until that's done, C12 MOM6 will only work with TinyBCs.

yvikhlya commented:
> @sanAkel I still do not see a `CF0012x6C_TM0072xTM0036` directory in:
> 
> ```
> /discover/nobackup/yvikhlia/coupled/Forcings/MOM6/
> ```
> 
> I think until that's done, C12 MOM6 will only work with TinyBCs.

All coupled bc files are moved to /discover/nobackup/projects/gmao/ssd/aogcm/atmosphere_bcs. I keep the old files at their old locations for now because people are running old experiments with them.

sanAkel commented:
> @sanAkel I still do not see a `CF0012x6C_TM0072xTM0036` directory in:
> 
> ```
> /discover/nobackup/yvikhlia/coupled/Forcings/MOM6/
> ```
> 
> I think until that's done, C12 MOM6 will only work with TinyBCs.

@mathomp4 I am not sure when and how @yvikhlya is handling all these ... let's please proceed with `TinyBCs` can @sdrabenh  work with `TinyBCs` ?

sdrabenh commented:
> can @sdrabenh work with `TinyBCs` ?

I am not familiar with running anything at C12. Whatever you guys decide is fine.

sanAkel commented:
> > can @sdrabenh work with `TinyBCs` ?
> 
> I am not familiar with running anything at C12. Whatever you guys decide is fine.

@sdrabenh you always ease the pain! 😄

mathomp4 commented:
> > @sanAkel I still do not see a `CF0012x6C_TM0072xTM0036` directory in:
> > ```
> > /discover/nobackup/yvikhlia/coupled/Forcings/MOM6/
> > ```
> > I think until that's done, C12 MOM6 will only work with TinyBCs.
> 
> All coupled bc files are moved to /discover/nobackup/projects/gmao/ssd/aogcm/atmosphere_bcs. I keep the old files at their old locations for now because people are running old experiments with them.

@yvikhlya The problem is that I don't think:
```
/discover/nobackup/projects/gmao/ssd/aogcm/atmosphere_bcs/Icarus-NLv3/MOM6/CF0012x6C_TM0072xTM0036 
```
has all the needed files. For example, when I try to run with that as the `GRIDDIR` I have five broken symlinks in the `scratch/` directory:

```
lrwxrwxrwx 1 mathomp4 g0620  126 Feb  2 15:45 SEAWIFS_KPAR_mon_clim.data -> /discover/nobackup/projects/gmao/ssd/aogcm/atmosphere_bcs/Icarus-NLv3/MOM6/CF0012x6C_TM0072xTM0036/SEAWIFS_KPAR_mon_clim.72x36
lrwxrwxrwx 1 mathomp4 g0620  115 Feb  2 15:45 MAPL_Tripolar.nc -> /discover/nobackup/projects/gmao/ssd/aogcm/atmosphere_bcs/Icarus-NLv3/MOM6/CF0012x6C_TM0072xTM0036/MAPL_Tripolar.nc
lrwxrwxrwx 1 mathomp4 g0620  112 Feb  2 15:45 vgrid.ascii -> /discover/nobackup/projects/gmao/ssd/aogcm/atmosphere_bcs/Icarus-NLv3/MOM6/CF0012x6C_TM0072xTM0036/vgrid50.ascii
...
lrwxrwxrwx 1 mathomp4 g0620  116 Feb  2 15:45 kmt_cice.bin -> /discover/nobackup/projects/gmao/ssd/aogcm/atmosphere_bcs/Icarus-NLv3/MOM6/CF0012x6C_TM0072xTM0036/cice/kmt_cice.bin
lrwxrwxrwx 1 mathomp4 g0620  117 Feb  2 15:45 grid_cice.bin -> /discover/nobackup/projects/gmao/ssd/aogcm/atmosphere_bcs/Icarus-NLv3/MOM6/CF0012x6C_TM0072xTM0036/cice/grid_cice.bin
```
These are in the TinyBCs set that @sanAkel provided me.

I then based the TinyBCs directory structure on that of:
```
/discover/nobackup/yvikhlia/coupled/Forcings/MOM6/CF0090x6C_TM0360xTM0210/
```
where there is a `cice/` directory and the other three files.

mathomp4 commented:
Note that this PR is still safe as long as anyone that wants to run C12 Newland MOM6, runs with my TinyBCs. It will still work fine for C90 OldLand MOM6.

sanAkel commented:
> Note that this PR is still safe as long as anyone that wants to run C12 Newland MOM6, runs with my TinyBCs. It will still work fine for C90 OldLand MOM6.

Thanks @mathomp4 Let me think about this a bit ...

yvikhlya commented:
@mathomp4 I split ocean and agcm bcs, since ocean bcs do not depend on agcm resolution and land model version, there is no need to copy them under every configuration. There is no GRIDDIR anymore, new locations are
```
/discover/nobackup/projects/gmao/ssd/aogcm/atmosphere_bcs
/discover/nobackup/projects/gmao/ssd/aogcm/ocean_bcs
```
I updated gcm_run.j and gcm_setup to work with new locations on the feature/aogcm branch, but this is not merged with develop yet.

mathomp4 commented:
> @mathomp4 I split ocean and agcm bcs, since ocean bcs do not depend on agcm resolution and land model version, there is no need to copy them under every configuration. There is no GRIDDIR anymore, new locations are
> 
> ```
> /discover/nobackup/projects/gmao/ssd/aogcm/atmosphere_bcs
> /discover/nobackup/projects/gmao/ssd/aogcm/ocean_bcs
> ```
> 
> I updated gcm_run.j and gcm_setup to work with new locations on the feature/aogcm branch, but this is not merged with develop yet.

@yvikhlya Ahh. Okay. If you don't mind, I can take a look at that branch and then merge the `GRIDDIR` to new dirs changes into my PR here?

yvikhlya commented:
> @yvikhlya Ahh. Okay. If you don't mind, I can take a look at that branch and then merge the `GRIDDIR` to new dirs changes into my PR here?

Oh yes, you are welcomed to do this. Instead of GRIDDIR, there are OBCS and ABCS variables now. Not sure if I did it the best way, so if you feel something needs to be tweaked, feel free to do this.

mathomp4 commented:
I'm blocking this until I can get in the updates from @yvikhlya. Hopefully unblocking tomorrow

yvikhlya commented:
> I'm blocking this until I can get in the updates from @yvikhlya. Hopefully unblocking tomorrow

What updates exactly?

mathomp4 commented:
> > I'm blocking this until I can get in the updates from @yvikhlya. Hopefully unblocking tomorrow
> 
> What updates exactly?

The replacement for `GRIDDIR`. I need to make sure I get the merge of changes over correctly. (since `feature/aogcm` still uses the >>><<< style instead of the @-style for options preferred by Ricardo).

yvikhlya commented:
> The replacement for `GRIDDIR`.

It is split to
```
/discover/nobackup/projects/gmao/ssd/aogcm/atmosphere_bcs
/discover/nobackup/projects/gmao/ssd/aogcm/ocean_bcs
```
Am I getting wrong something?

mathomp4 commented:
> > The replacement for `GRIDDIR`.
> 
> It is split into
> 
> ```
> /discover/nobackup/projects/gmao/ssd/aogcm/atmosphere_bcs
> /discover/nobackup/projects/gmao/ssd/aogcm/ocean_bcs
> ```
> 
> Am I getting wrong something?
> 
> EDIT: Note, that these directories are not copied to NAS currently. EDIT2: Matt, it looks like you get everything right in your commit. What I could be missing?

Yury,

I found some files are missing at NCCS. I pinged on Teams, but for history's sake:

1. Some updates are needed to your atmosphere_bcs directory for Icarus Old Land C90 TM360. The model is expecting files like visdf_90x540.dat but they aren't there. These are all broken symlinks:
```
visdf.dat -> /discover/nobackup/projects/gmao/ssd/aogcm/atmosphere_bcs/Icarus/MOM6/CF0090x6C_TM0360xTM0210/visdf_90x540.dat
nirdf.dat -> /discover/nobackup/projects/gmao/ssd/aogcm/atmosphere_bcs/Icarus/MOM6/CF0090x6C_TM0360xTM0210/nirdf_90x540.dat
vegdyn.data -> /discover/nobackup/projects/gmao/ssd/aogcm/atmosphere_bcs/Icarus/MOM6/CF0090x6C_TM0360xTM0210/vegdyn_90x540.dat
lai.data -> /discover/nobackup/projects/gmao/ssd/aogcm/atmosphere_bcs/Icarus/MOM6/CF0090x6C_TM0360xTM0210/lai_clim_90x540.data
green.data -> /discover/nobackup/projects/gmao/ssd/aogcm/atmosphere_bcs/Icarus/MOM6/CF0090x6C_TM0360xTM0210/green_clim_90x540.data
```
I think your directory only has them without the "90x540"

2.  I think the NLv3 C12 ocean_bcs has some missing files. These two files are in my TinyBCs from Santha, but not in your directory:
```
/gpfsm/dnb44/mathomp4/TinyBCs-GitV10/ocean_bcs/MOM6/72x36/INPUT/grid_spec.nc
/gpfsm/dnb44/mathomp4/TinyBCs-GitV10/ocean_bcs/MOM6/72x36/INPUT/ocean_mosaic.nc
```

yvikhlya commented:
@mathomp4 Could you check again? Let me know if anything is still missing.

mathomp4 commented:
> @mathomp4 Could you check again? Let me know if anything is still missing.

@yvikhlya I think we got it. It seems to work for me. I also fixed up TinyBCs to, I think, work with the new `ABCSDIR` and `OBCSDIR`. Though testing will find that out...

mathomp4 commented:
Sigh. Or not. Dang it. I messed up something...

mathomp4 commented:
@yvikhlya I think one more directory still missing. When I try to run C90 360x210 (which is my "usual" MOM6 run) I see these broken symlinks:
```
kmt_cice.bin -> /discover/nobackup/projects/gmao/ssd/aogcm/ocean_bcs/MOM6/360x210/cice/kmt_cice.bin
grid_cice.bin -> /discover/nobackup/projects/gmao/ssd/aogcm/ocean_bcs/MOM6/360x210/cice/grid_cice.bin
```
Indeed, the `360x210` case isn't in `ocean_bcs`:
```
$ ls -l /discover/nobackup/projects/gmao/ssd/aogcm/ocean_bcs/MOM6/
total 4
drwxr-xr-x 6 yvikhlia g0609 4096 Sep 16 12:12 1440x1080/
drwxr-xr-x 6 yvikhlia g0609 4096 Jan 31 13:01 1440x1080_newtopo/
drwxr-xr-x 6 yvikhlia g0609 4096 Oct 13 12:19 72x36/
-rw-r--r-- 1 yvikhlia g0609 3160 Jan 20 12:01 generate_topo.md
```

yvikhlya commented:
@mathomp4 This is because MOM6 c90x360x210 does not work with new land BCs and I thought we are going to abandon it. But I can copy it to new location to use with old land BCs, hold on.

mathomp4 commented:
> @mathomp4 This is because MOM6 c90x360x210 does not work with new land BCs and I thought we are going to abandon it. But I can copy it to new location to use with old land BCs, hold on.
> 
> EDIT: Done.

@yvikhlya Thanks. I'll retest as soon as I get nodes.

And, yes, it is sort of "abandoned", but I can only really run two cases now. C90 Oldland and C12 Newland. Probably good to keep the old girl going until I can transition to another "mid-resolution" case that you or @sanAkel can provide me. 😄

sanAkel commented:
> > @mathomp4 This is because MOM6 c90x360x210 does not work with new land BCs and I thought we are going to abandon it. But I can copy it to new location to use with old land BCs, hold on.
> > EDIT: Done.
> 
> @yvikhlya Thanks. I'll retest as soon as I get nodes.
> 
> And, yes, it is sort of "abandoned", but I can only really run two cases now. C90 Oldland and C12 Newland. Probably good to keep the old girl going until I can transition to another "mid-resolution" case that you or @sanAkel can provide me. 😄

@mathomp4 There are a few things already queued/piled up. Building about a `1-deg` ocean case is one of them, but not at the _top_ right now. However, the `0.25-deg` case is, which is (from a _non oceanographer_ perspective) at the "high" end of resolution spectrum! Meanwhile, enjoy the `C12 x 5-deg`!

mathomp4 commented:
Removing the blocker. It looks like all is well with this. C12 runs. @bena-nasa and I need to do some BCs work at NAS, but I think we are good.

mathomp4 commented:
Closing in favor of #292
